### PR TITLE
Add changelog appendix

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

**Comment by pietroalbini** [comment]
bors try

File: src/changelog.rst
Comment (pietroalbini) [inline src/changelog.rst:77]: Re-reading the change to reword this section, it feels wrong that we removed the description of library crates in the FLS. Can you send a follow-up PR to add back that paragraph and glossary entry?

File: src/changelog.rst
Comment (pietroalbini) [inline src/changelog.rst:85]: This feels like a gap in the FLS we should fill (and describe).

File: src/changelog.rst
Comment (pietroalbini) [inline src/changelog.rst:146]: The description of what changed here is missing.

**Review by pietroalbini — CHANGES_REQUESTED** [review]
Pushed some commits to reword and reformat the changelog entries a bit. I also configured the lint to prevent having paragraph IDs in this page, which just add noise for no benefit.

Left comments on things that I feel we need to expand on.

**Comment by pietroalbini** [comment]
bors try

**Comment by pietroalbini** [comment]
bors try

**Review by pietroalbini — APPROVED** [review]
Marking Lukas's changes as approved, waiting for Lukas to approve my changes before we merge this.

I think the best thing is to merge this PR right now, and tackle the three points I mentioned in https://github.com/ferrocene/specification/pull/511#pullrequestreview-2158659401 in followup PRs that both fix the points and update the changelog.

**Review by Veykril — COMMENTED** [review]
👍 (cant approve my own PR)

**Comment by pietroalbini** [comment]
bors merge

**Pull request merged**
